Cleanup sidebar to use flexbox

xPaw opened the pull request:
First commit removes `div` wrapper added in 0404671 which is not necessary.

The second commit cleans up channels to use flexbox instead of floats. Also fixes up join channel form to use same padding as channels.

These changes should make it easier to further improve the sidebar (e.g. for mobile).

astorije commented:
Very cool!
I noticed you have some alignment issues:

Before | After
--- | ---
<img width="400" alt="before" src="https://user-images.githubusercontent.com/113730/36952486-dbfc7dda-1fde-11e8-96de-6ac8b69d4524.png"> | <img width="251" alt="after" src="https://user-images.githubusercontent.com/113730/36952489-dc22668a-1fde-11e8-9cb9-9f42ee984dde.png"><br><kbd>+</kbd> icon should be closer to the right border to match <kbd>⌄</kbd>'s distance to the left border.
<img width="251" alt="before2" src="https://user-images.githubusercontent.com/113730/36952487-dc0999f2-1fde-11e8-9a55-913e18705ebf.png"> | <img width="251" alt="after2" src="https://user-images.githubusercontent.com/113730/36952488-dc165548-1fde-11e8-9bdc-3191e746c053.png"><br>Form should be centered in the sidebar.

dgw commented:
The join form looks better to me in the after image. Lining up the edges of the form with the edges of the channel list works really well.

astorije commented:
It feels weird and uncomfortable to me that the form is not visually centered. Unequal margins are just killing me :)

xPaw commented:
@astorije What I don't like about making the right side match the padding of the arrow is that unread badges look very unbalanced:

![05-090457051](https://user-images.githubusercontent.com/613331/36966304-042fbe80-2065-11e8-9e42-e3f7c31f3c05.png)

xPaw commented:
Okay I replaced the network icon with the arrow, and moved network status icon separately to the right (and added tooltips).

The only problem I have is that collapsed arrow is changing dimensions due to the transform, and I don't seem to find a solution to that.

astorije commented:
This is really nice 👏 I love that we can have tooltips on the icons now!

A few things I've noticed:

- Since we're nitpicking on margins, you could remove a bit of the left margin of the arrow, so it's centered between the left side of the window and the network name (removing 5px to both width and padding-left of `#sidebar .network .collapse-network` seems to do the job, so ends up being respectively 39px and 10px). If you also do this on the right-hand side to balance, that's an extra 10px for channel/network names 🤷‍♂️
  If unclear, I am talking of (currently):
  <img width="127" alt="screen shot 2018-03-05 at 23 21 01" src="https://user-images.githubusercontent.com/113730/37014211-ff9b65aa-20cc-11e8-97af-0d84f98a4fa9.png">

- Caret now appears on empty networks.

- Hovering the caret also highlights the server. The opposite works fine, i.e. hovering the server does not highlight the caret.

- Do you want to also get rid of the server icon here: <img width="108" alt="screen shot 2018-03-05 at 23 28 28" src="https://user-images.githubusercontent.com/113730/37014213-0a02288a-20cd-11e8-99a8-3ff3b92fdc37.png"> (Could use one less problematic icon for Font Awesome v5 transition 😅)

- It looks like channels get double top margin when the join form is opened, not sure if this is intentional or not.

xPaw commented:
> you could remove a bit of the left margin of the arrow

I centered the arrow on the channel icons below it.

> Caret now appears on empty networks.

Intentionally, otherwise hiding it leaves big empty space.

> Hovering the caret also highlights the server. The opposite works fine, i.e. hovering the server does not highlight the caret.

I'm not sure how I could fix this this CSS.

> Do you want to also get rid of the server icon here:

Nah. It's used in the context menu too.

> It looks like channels get double top margin when the join form is opened

You mean the margin after the join button? That was indeed intentional. I can reduce it.

astorije commented:
> I centered the arrow on the channel icons below it.

Right, but the channel icons themselves are not centered in the space around them (something like `padding: 8px 14px;`ish instead of `padding: 8px 18px;` L571 would fix that). No big deal though.

> Intentionally, otherwise hiding it leaves big empty space.

I really think this should go. There is no distinction between having no channels or a collapsed network, and the fact that the caret can be clicked and have both states even though there is nothing collapsed/expanded is weird.
I don't mind about the big empty space because it's a temporary/abnormal state that is going to be solved as soon as a channel is joined, but having a button that's effectively not functional is odd.

> Nah. It's used in the context menu too.

I'd be okay if we remove it from there too, but that was more of a thought. We'll rethink icons when we upgrade to FA v5 anyway.

Nice job on the other fixes!

xPaw commented:
> Right, but the channel icons themselves are not centered in the space around them

Do you mind making a commit to fix this up yourself?

> I don't mind about the big empty space because it's a temporary/abnormal state that is going to be solved as soon as a channel is joined, but having a button that's effectively not functional is odd.

It is indeed an abnormal state, which is why I'm not keen on writing more CSS to cover this case (hide it, and fix empty space, as the arrow isn't absolutely positioned anymore). I think displaying the arrow is completely fine, and is better than the empty space.

astorije commented:
> Do you mind making a commit to fix this up yourself?

Not at all, done in cb1ad42c, let me know what you think.

> I'm not keen on writing more CSS to cover this case [...] I think displaying the arrow is completely fine, and is better than the empty space.

I very much disagree with this, it simply looks broken at the moment:

![broken-collapse](https://user-images.githubusercontent.com/113730/37125267-d4efaabe-2239-11e8-8e7a-274113aeef1e.gif)

And there isn't much more CSS to write, @McInkay already did it, there was just a bit of adjustment to do.
We should really fix that. I made you 2 versions, you just have to pick one :)

**First version:**

<img width="231" alt="screen shot 2018-03-07 at 18 58 04" src="https://user-images.githubusercontent.com/113730/37125287-ee2cea28-2239-11e8-8cb8-4a14f7b1773b.png">


```diff
diff --git a/client/css/style.css b/client/css/style.css
index 09e13bf1..fde50569 100644
--- a/client/css/style.css
+++ b/client/css/style.css
@@ -723,6 +723,11 @@ kbd {
 	transform: rotate(45deg) translateZ(0);
 }

+#sidebar .network .lobby:first-child:nth-last-child(2) .collapse-network {
+	/* Hide collapse button if there are no channels/queries */
+	visibility: hidden;
+}
+
 #sidebar .network .collapse-network {
 	width: 40px;
 	opacity: 0.4;
```

**Second version:**

<img width="233" alt="screen shot 2018-03-07 at 18 57 42" src="https://user-images.githubusercontent.com/113730/37125291-f0d7faec-2239-11e8-969e-998a6090e23f.png">

```diff
$ git --no-pager diff
diff --git a/client/css/style.css b/client/css/style.css
index 09e13bf1..c22a569f 100644
--- a/client/css/style.css
+++ b/client/css/style.css
@@ -723,6 +723,12 @@ kbd {
 	transform: rotate(45deg) translateZ(0);
 }

+#sidebar .network .lobby:first-child:nth-last-child(2) .collapse-network {
+	/* Hide collapse button if there are no channels/queries */
+	width: 0;
+	overflow: hidden;
+}
+
 #sidebar .network .collapse-network {
 	width: 40px;
 	opacity: 0.4;
```

I'll let you choose, anything is better than a broken button.
You'll notice I even made them in patch format in case you just want to apply yourself (and discovered `git --no-pager` along the way), but I can also make the commit myself, up to you! :)

astorije commented:
(btw, the force-push was just me rebasing with `master` to test, no changes made to your commits)

McInkay commented:
Fwiw, I agree with @astorije

And I prefer the empty space one, don't like it expanding to fill the space, looks awkward. And I don't think it massively matters that we have a bit of empty space because, as you both say, it's not like this is a very common thing to have, a server with no channels.

xPaw commented:
OK, I will hide the button if there are no channels. Using `visibility: hidden` the button is still clickable by the way.
